galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunResultRas.java

```diff
@@ -80,16 +83,17 @@ private RunResult getRun(String id) throws ResultArchiveStoreException {
        IRunResult run = null;
        
         for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
-             
+           
            run = directoryService.getRunById(id);
+           break;
```

Comment:
if (run != null) then break;   !!!!!!!!
Come on Charlie, I know you can do better